nativesoundsystem/src/main/jni/aaudio/AAudioManager.cpp

```diff
@@ -97,29 +119,31 @@ bool AAudioManager::createEngine(SoundSystem *soundSystem) {
 #endif
     engine.bitsPerSample_ = SampleFormatToBpp(engine.sampleFormat_);
 
-    StreamBuilder builder(engine.sampleChannels_,
-                          engine.sampleFormat_,
-                          AAUDIO_SHARING_MODE_SHARED,
-                          AAUDIO_DIRECTION_OUTPUT);
+    StreamBuilder builder;
+    engine.playStream_ = builder.CreateStream(
+            engine.sampleFormat_,
+            engine.sampleChannels_,
+            AAUDIO_SHARING_MODE_SHARED,
+            AAUDIO_DIRECTION_OUTPUT,
+            INVALID_AUDIO_PARAM,
+            dataCallback,
+            &engine);
 
-    engine.playStream_ = builder.Stream();
     assert(engine.playStream_);
 
     PrintAudioStreamInfo(engine.playStream_);
-
     engine.sampleRate_ = AAudioStream_getSampleRate(engine.playStream_);
+    engine.framesPerBurst_ = AAudioStream_getFramesPerBurst(engine.playStream_);
+    engine.defaultBufSizeInFrames_ = AAudioStream_getBufferSizeInFrames(engine.playStream_);
+    AAudioStream_setBufferSizeInFrames(engine.playStream_, engine.framesPerBurst_);
+    engine.bufSizeInFrames_ = engine.framesPerBurst_;
+
     aaudio_result_t result = AAudioStream_requestStart(engine.playStream_);
```

Comment:
Insert this code after ```AAudioStream_requestStart``` :
```java
aaudio_stream_state_t inputState = AAUDIO_STREAM_STATE_STARTING;
aaudio_stream_state_t nextState = AAUDIO_STREAM_STATE_UNINITIALIZED;
result = AAudioStream_waitForStateChange(engine.playStream_, inputState, &nextState, 1000);
```
to wait for stream state change. Before, state was always 'starting', maybe a bug, but with ```AAudioStream_waitForStateChange```, we can wait the new state which is 'started'. Behavior doesn't change, we can still pay sound, but maybe, in another update of AAudio, it will have an impact.